### PR TITLE
Improving installation

### PR DESCRIPTION
**Pull request by mehdisadeghi**
This PR addresses a few issues in #18 and #21 and some minor improvements:
1. The installation instructions are improved to explicitly cover installation of required header files.
2. External installation script changed to allow users to specify a custom location for installation
3. JYU-LB added as a sub-module with corresponding document update.
4. A requirements.txt (non-dev) file added along with corresponding document update.
5. Fixes travis links.

### CODE REVIEW

File: JYU-LB [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit c7d07ba1666966f7c1cf5a8202d078240cf43ab4
```
Comment (nathanfranklin) [inline JYU-LB:1]: A tagged release of JYU-LB should be used.  @kemattil , can you tag a new release.

**Comment by nathanfranklin** [comment]
Looks good.  I just left one comment  Thanks @mehdisadeghi

File: JYU-LB (hunk 1 above)
Comment (mehdisadeghi) [inline JYU-LB:1]: I just made a release 0.1.1 for JYU-LB which points to this commit. Git submodule points exactly to one commit on the remote branch.

**Comment by nathanfranklin** [comment]
@kemattil , @tuopuu ,  can you take a look at this PR

**Comment by kemattil** [comment]
@nathanfranklin, I tagged JYU-LB (0.1.2) according to the instructions by Ioannis: there is now a related PR in JYU-LB repo (branch v0.1.2-release).

File: README.rst [hunk 2]
```diff
@@ -5,34 +5,50 @@ Internal and file-IO based JYU-LB engine-wrappers for the SimPhoNy framework.
...
   simphony-jyulb treats the BC data component as a dummy variable (currently there
-  is no SimPhoNy definition for BC).  
+  is no SimPhoNy definition for BC).
 
 Installation
 ------------
 
-The package requires python 2.7.x. Installation is based on setuptools::
+The package requires python 2.7.x, cython and JYU-LB header files along with
+`simphony-common` package to build and install cython extension. To get the
```
Comment (kemattil) [inline README.rst:42]: Why only the JYU-LB header files? I mean, for compiling the cython extensions the JYU-LB source files are needed ...

Comment (nathanfranklin) [inline README.rst:42]: @mehdisadeghi

**Comment by nathanfranklin** [comment]
@kemattil ,   @mehdisadeghi has added another commit based on your comment.

**Comment by kemattil** [comment]
Seems ok to me! :+1:

**Comment by nathanfranklin** [comment]
Great.  Can you merge and make a release.  Thanks

**Pull request merged**